lib/logthrdestdrv.h

```diff
@@ -116,7 +119,13 @@ struct _LogThreadedDestDriver
   gint workers_started;
 
   gint stats_source;
-  gint32 seq_num;
+
+  /* this counter is not thread safe if there are multiple worker threads,
+   * in that case, one needs to use LogThreadedDestWorker->seq_num, which is
+   * static for a single insert() invocation, whereas this might be
+   * increased in parallel by the multiple threads. */
+
+  gint32 shared_seq_num;
```

Comment:
Shouldn't we use a 64 bit atomic counter (`atomic-gsize`) for `shared_seq_num`?
It will overflow after ~3hours when the message rate is 400k/sec.
Or it's not a problem as it is used with timestamp?  and (currently :) ) we cannot do 2^32-1 msg/sec...

Reply:
It's only meant to be used for ordering and is explicitly sized as 32 bits:

quoting rfc5424:

7.3.1.  sequenceId

   The "sequenceId" parameter tracks the sequence in which the
   originator submits messages to the syslog transport for sending.  It
   is an integer that MUST be set to 1 when the syslog function is
   started and MUST be increased with every message up to a maximum
   value of 2147483647.  If that value is reached, the next message MUST
   be sent with a sequenceId of 1.